### ISSUE TITLE
Allow calc to use fields from both layers in a -join

### ISSUE THREAD

**Issue by yuletide** [mbloch/mapshaper#536]
Currently I'm joining two datasets, POINTS (bands) and POLY (admin units with population, defined as UN_2020_E), and would love to calculate both the sum, and per capita statistics in one go. 

The command I'm using: 

```
mapshaper -i poly.shp\
         -join points.shp \
         calc 'bands_count = count(), bands_per = bands_count / UN_2020_E' \
         -o output/percapita.shp
```
This currently errors: `[join] ReferenceError in expression [bands_count = count(), bands_per = bands_count / UN_2020_E]: UN_2020_E is not defined`

Is this an error in my calc or just not supported? I'd love to do this all in one go vs running another command after to calculate the final field.

**Comment by mbloch** [comment]
You're right, there is no way to reference attribute data for the target layer in a `calc` expression. You would have to add an `-each` command after the join, like this: `-each 'bands_per = bands_count / UN_2020_E'`.

Currently, only source fields are available in `-join calc` because I found that this simplified the expression syntax and was generally sufficient. I'll consider adding destination fields if we can find a clear and intuitive syntax for referencing those variables. Mapshaper uses "target" and "source" to refer to the two layers being joined... maybe those names could be exposed in the `calc=` expression, like this:

```
-join points.shp calc 'bands_count = count(), bands_per = bands_count / target.UN_2020_E'
```

Do you have a better suggestion?

**Comment by yuletide** [comment]
That worked great! I think the -each is a great stopgap, and could be presented as "the way to do this" in the docs for now. For more direct join support the `target` syntax makes sense to me also. It could be interesting to try to guess the field names on first pass, i.e. if the field name is not available in the source data, check the target data. But clearly runs into issues when field names are the same.
